examples/nature_of_code/chp_01_vectors/1_2_bouncingball_vectors_object.rs

```diff
@@ -0,0 +1,79 @@
+// The Nature of Code
+// Daniel Shiffman
+// http://natureofcode.com
+//
+// Example 1-2: Bouncing Ball, with Vector!
+extern crate nannou;
+
+use nannou::prelude::*;
+
+fn main() {
+    nannou::run(model, event, view);
+}
+
+struct Model {
+    ball: Ball,
+}
+
+struct Ball {
+    position: Point2<f32>,
+    velocity: Vector2<f32>,
+}
+
+impl Ball {
+    fn new() -> Self {
+        let position = Point2::new(100.0, 100.0);
+        let velocity = Vector2::new(2.5, 5.0);
+        Ball { position, velocity }
+    }
+
+    fn update(&mut self, rect: Rect<f32>) {
+        // Add the current speed to the position.
+        self.position += self.velocity;
+
+        if (self.position.x > rect.right()) || (self.position.x < rect.left()) {
+            self.velocity.x = self.velocity.x * -1.0;
+        }
+        if (self.position.y > rect.top()) || (self.position.y < rect.bottom()) {
```

Comment:
Don't need parentheses around these greater than or less than checks - doesn't really matter but just thought I'd mention it!